### PR TITLE
feat!: make `esbuild` a peer dependency where possible

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
Supersedes @gabro's #1018

### CODE REVIEW

File: packages/esbuild-plugin-next/package.json [hunk 1]
```diff
@@ -16,9 +16,12 @@
...
   "license": "MIT",
   "dependencies": {
-    "@vanilla-extract/integration": "^6.0.2"
+    "@vanilla-extract/integration": "^6.2.2"
   },
   "devDependencies": {
     "esbuild": "0.17.6"
+  },
+  "peerDependencies": {
+    "esbuild": "^0.17.0 || ^0.18.0 || ^0.19.0"
```
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply packages/esbuild-plugin-next/package.json:25]: Updating the peer dep gives you an explicit change to add tests for it though, which is a huge win imo

And if people don't complain about support, you don't have to bother with it either until you want it yourself 🤷‍♂️

[the remark this answers is not in the document]
Comment (AndrewLeedham) [reply packages/esbuild-plugin-next/package.json:25]: Yeah I can definitely see the advantage. Peer dependencies are just one of those things that can get in the way, if the package falls out of maintenance for example.

**Review by AndrewLeedham — APPROVED** [review]

File: packages/esbuild-plugin-next/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (MichaelDeBoey) [reply packages/esbuild-plugin-next/package.json:25]: > if the package falls out of maintenance

You're talking about `esbuild` or `@vanilla-extract/*` here?

**Comment by Benji1** [comment]
Hey @AndrewLeedham,
we have other dependencies in our repo that uses esbuild 0.19.4 and esbuild's `Plugin` type has changed since 0.17.6.
So this PR would help us very much, I can resolve the conflicts if it would help to get this in :smile:

**Comment by AndrewLeedham** [comment]
> Hey @AndrewLeedham, we have other dependencies in our repo that use esbuild 0.19.4 and esbuild's `Plugin` type has changed since 0.17.6. So this PR would help us very much, I can resolve the conflicts if it would help to get this in 😄

@Benji1 This is not my PR. Other than providing some feedback, I am not involved in this, and cannot move it forward.

**Comment by askoufis** [comment]
Thanks for moving the needle on this one. This has been superseded by #1232. We decided to go with an approach similar to what @gabro tried, but making `esbuild` an optional peer dependency. If you still have problems regarding esbuild versions, please raise an issue.

**Pull request closed without merging**